Sources/NIOCore/Docs.docc/swift-concurrency.md

```diff
@@ -29,99 +33,115 @@ promise.completeWithTask {
+into Swift Concurrency. This can be done by wrapping a ``Channel`` via the
+`NIOAsyncChannel/init(synchronouslyWrapping:configuration:)`
+initializer. Under the hood, this initializer adds two channel handlers to the
+end of the channel's pipeline. These handlers bridge the read and write side of
+the channel. Additionally, the handlers work together to close the channel once
+both the reading and the writing have finished.
 
-
-This is how you can wrap an existing channel into a `NIOAsyncChannel`, consume the inbound data and
-echo it back outbound.
+This is how you can wrap an existing channel into a `NIOAsyncChannel`, consume
+the inbound data and echo it back outbound.
 
 ```swift
 let channel = ...
-let asyncChannel = try NIOAsyncChannel(synchronouslyWrapping: channel, inboundType: ByteBuffer.self, outboundType: ByteBuffer.self)
+let asyncChannel = try NIOAsyncChannel<ByteBuffer, ByteBuffer>(synchronouslyWrapping: channel)
 
 for try await inboundData in asyncChannel.inboundStream {
     try await asyncChannel.outboundWriter.write(inboundData)
 }
 ```
 
-The above code works nicely; however, you must be very careful at what point you wrap your channel
-otherwise you might lose some reads. For example your channel might be created by a `ServerBootstrap`
-for a new inbound connection. The channel might start to produce reads as soon as it registered its
-IO which happens after your channel initializer ran. To avoid potentially losing reads the channel
-must be wrapped before it registered its IO.
-Another example is when the channel contains a handler that does protocol negotiation. Protocol negotiation handlers
-are usually waiting for some data to be exchanged before deciding what protocol to chose. Afterwards, they
-often modify the channel's pipeline and add the protocol appropriate handlers to it. This is another
-case where wrapping of the `Channel` into a `NIOAsyncChannel` needs to happen at the right time to avoid
-losing reads.
+The above code works nicely; however, you must be very careful at what point you
+wrap your channel otherwise you might lose some reads. For example your channel
+might be created by a `ServerBootstrap` for a new inbound connection. The
+channel might start to produce reads as soon as it registered its IO which
+happens after your channel initializer ran. To avoid potentially losing reads
+the channel must be wrapped before it registered its IO. Another example is when
+the channel contains a handler that does protocol negotiation. Protocol
+negotiation handlers are usually waiting for some data to be exchanged before
+deciding what protocol to chose. Afterwards, they often modify the channel's
+pipeline and add the protocol appropriate handlers to it. This is another case
+where wrapping of the `Channel` into a `NIOAsyncChannel` needs to happen at the
+right time to avoid losing reads.
 
-### Async bootstrap
+### New typed asynchronous bootstraps
```

Comment:
This is a bit misleading: they're not new bootstraps, they're new `async` methods on the existing bootstraps.